### PR TITLE
Small fixes for position control

### PR DESCRIPTION
**Pull request by Ratoone**
Made the path planner optimize the branching order, minimizing localized collisions close to the target and stop charging toward the target if a path is not found

### CODE REVIEW

File: src/control/positionControl/PositionControl.cpp [hunk 1]
```diff
@@ -11,11 +12,17 @@ RobotCommand PositionControl::computeAndTrackPath(const world::Field &field, int
...
     // if the target position is outside of the field (i.e. bug in AI), do nothing
-    if(!collisionDetector.isPointInsideField(targetPosition)){
+    if (!collisionDetector.isPointInsideField(targetPosition)) {
+        RTT_WARNING("Target point not in field for robot ID ", robotId)
         return {};
     }
 
     if (shouldRecalculatePath(currentPosition, targetPosition, currentVelocity, robotId)) {
+        // if the robot is close to the final position and can't get there, stop
+        if (!computedPaths[robotId].empty() && (currentPosition - targetPosition).length() < 4 * Constants::ROBOT_RADIUS()) {
```
Comment (HaicoDorenbos) [inline src/control/positionControl/PositionControl.cpp]: 4 * Constant::ROBOT_RADIUS is a magical value

File: src/control/positionControl/pathPlanning/NumTreesPlanning.cpp [hunk 2]
```diff
@@ -34,12 +34,16 @@ NumTreesPlanning::computePath(const Vector2 &robotPosition, const Vector2 &targe
...
             if (parentCollision){
-                auto branches = branchPath(*point.getParent(), parentCollision.value());
+                auto branches = branchPath(*point.getParent(), parentCollision.value(), targetPosition);
                 std::for_each(branches.begin(), branches.end(), [&pointQueue](PathPointNode& newPoint){
                     pointQueue.push(newPoint);
                     });
                 continue;
             }
+            // current path has no collisions - update current best path
+            if (point.getPosition() - targetPosition < finalPath.getPosition() - targetPosition){
```
Comment (HaicoDorenbos) [inline src/control/positionControl/pathPlanning/NumTreesPlanning.cpp:44]: What? I did not know that comparisons are defined for Vector2?

**Review by HaicoDorenbos — APPROVED** [review]
Beside 1 magical value, everything looks fine to me :+1:

File: src/control/positionControl/pathPlanning/NumTreesPlanning.cpp (hunk 2 above)
Comment (Ratoone) [reply src/control/positionControl/pathPlanning/NumTreesPlanning.cpp:44]: The < operator is defined. It compares the length of 2 vectors

**Pull request merged**